engines/agi/saveload.cpp

```diff
@@ -85,11 +85,19 @@ int AgiEngine::saveGame(const Common::String &fileName, const Common::String &de
 
 	out->writeUint32BE(AGIflag);
 
+	const char *descriptionStringC;
+	if (_game._vm->getLanguage() != Common::HE_ISR) {
+		descriptionStringC = descriptionString.c_str();
+	} else {
+		Common::U32String u32string = descriptionString.decode(Common::kWindows1255);
+		descriptionStringC = u32string.encode(Common::kUtf8).c_str();
+	}
+
 	// Write description of saved game, limited to SAVEDGAME_DESCRIPTION_LEN characters + terminating NUL
 	char description[SAVEDGAME_DESCRIPTION_LEN + 1];
 
 	memset(description, 0, sizeof(description));
-	strncpy(description, descriptionString.c_str(), SAVEDGAME_DESCRIPTION_LEN);
+	strncpy(description, descriptionStringC, SAVEDGAME_DESCRIPTION_LEN);
```

Comment:
And here: `description->c_str()`

Reply:
I assume that this depends on your previous suggestion.
If I understand correctly, it's not relevant with the fix that I've done.